### PR TITLE
Adds riiif cache management.

### LINKED ISSUE #1387: Configure cache management for RIIIF
### Descriptive summary

We are going to need some sort of cron script to purge derivatives created by riiif on our web/application server.

### Expected behavior
Derivates get purged on some time frame.

### Actual behavior

Derivates continue to stay in tmp folder.

### PR DESCRIPTION
**Pull request by scherztc**
Fixes #1387 

Present short summary (50 characters or less)

Sets the riiif cache duration for 90 days.
Sets the riiif cache location to tmp/network_files
Adds a cache clean up script.  (parameters: size, duration, and log output)

Description can have multiple paragraphs and you can use code examples inside:

config/initializers/riiif.rb

### CODE REVIEW

File: config/initializers/riiif.rb [hunk 1]
```diff
@@ -26,4 +26,5 @@ def logger
 
 # Riiif::Image.authorization_service = IIIFAuthorizationService
 
-Riiif::Engine.config.cache_duration_in_days = 365
+Riiif::Engine.config.cache_duration_in_days = 90
+Riiif::Image.file_resolver.cache_path = 'tmp/network_files'
```
Comment (hortongn) [inline config/initializers/riiif.rb]: @scherztc If we keep the cache in tmp/network_files, the cache will get blown away every time we deploy.  I think we should make this directory be `/srv/apps/scholar-riiif-cache` (outside of our app directory).  Calling it scholar-riiif-cache will make it more identifiable than "network_files".

File: script/riiif_cache_clean.sh [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+# riiif_cache_clean.sh
+# Cron script for maintaining the loris cache size.
+#
+# CAUTION - This script deletes files. Be careful where you point it!
+#
+
+LOG="../tmp/riiif_cache_clean.log"
+
+# Check that the cache directories...
+IMG_CACHE_DIR="../tmp/network_files"
```
Comment (hortongn) [inline script/riiif_cache_clean.sh]: @scherztc Change this to `/srv/apps/scholar-riiif-cache` too

File: script/riiif_cache_clean.sh [hunk 3]
```diff
@@ -0,0 +1,64 @@
+#!/bin/bash
+
+#
+# riiif_cache_clean.sh
+# Cron script for maintaining the loris cache size.
+#
+# CAUTION - This script deletes files. Be careful where you point it!
+#
+
+LOG="../tmp/riiif_cache_clean.log"
```
Comment (hortongn) [inline script/riiif_cache_clean.sh]: @scherztc Let's put the log in our log directory so we don't lose it every time we deploy.  So `log/riiif_cache_clean.log`

File: script/riiif_cache_clean.sh [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+
+# Check that the cache directories...
+IMG_CACHE_DIR="../tmp/network_files"
+
+# ...is below a certain size...
+REDUCE_TO=0 #0 gb
```
Comment (hortongn) [inline script/riiif_cache_clean.sh]: @scherztc if REDUCE_TO is set to zero, this script does nothing more than just wipe out everything in the cache directory every time we run it.  So I suggest setting it to 500000 for now so the cache never gets above half a gigabyte.  The most recent cache files up to half a gig will stay in the directory after we run the script.

**Comment by scherztc** [comment]
@hortongn :  Should this be a bamboo variable?  That way we could use the local_copy script to point it back to tmp for our local dev environments?  Do we then make it part of the environment puppet script to create those folder?

**Comment by hortongn** [comment]
@scherztc Other than the suggested changes I mentioned above, this script does what it should.  I'll merge after those tweaks.

**Comment by hortongn** [comment]
@scherztc I think you're right about `Riiif::Image.file_resolver.cache_path` needing to be a bamboo variable.  We want that cache directory to be in a different place on our servers than in our local environments.

1. Take the existing config/initializers/riiif.rb as it is and rename it to config/initializers/riiif.rb.local
1. copy config/initializers/riiif.rb.local to config/initializers/riiif.rb.bamboo
1. Edit config/initializers/riiif.rb.bamboo and stick a bamboo variable in the right spot.
1. Edit script/copy_config_local.sh and script/copy_config_bamboo.sh and add a line at the bottom of both

**Comment by hortongn** [comment]
@scherztc One final change to this PR:  Now that you have created `config/initializers/riiif.rb.bamboo` and `config/initializers/riiif.rb.sample`, you can do a `git rm config/initializers/riiif.rb` to remove that file from our repo.  It will be created when someone runs copy_config_local.sh or copy_config_bamboo.sh

File: script/copy_config_bamboo.sh [hunk 5]
```diff
@@ -13,3 +13,4 @@ cp /srv/apps/curate/curate_uc-STAGE/config/initializers/devise.rb.bamboo /srv/ap
 cp /srv/apps/curate/curate_uc-STAGE/config/environments/development.rb.bamboo /srv/apps/curate/curate_uc-STAGE/config/environments/development.rb
 cp /srv/apps/curate/curate_uc-STAGE/config/environments/production.rb.bamboo /srv/apps/curate/curate_uc-STAGE/config/environments/production.rb
 cp /srv/apps/curate/curate_uc-STAGE/config/authentication.yml.bamboo /srv/apps/curate/curate_uc-STAGE/config/authentication.yml
+cp /srv/apps/curate/curate_uc-STAGE/config/initializers/riiif.rb.bamboo cp /srv/apps/curate/curate_uc-STAGE/config/initializers/riiif.rb
```
Comment (hortongn) [inline script/copy_config_bamboo.sh]: @scherztc Once more fix needed.  There is an extra `cp` in the middle of this line that needs to be removed.

**Comment by scherztc** [comment]
@hortongn Ready to merge.

File: script/copy_config_bamboo.sh (hunk 5 above)
Comment (scherztc) [reply script/copy_config_bamboo.sh]: @hortongn -  This is ready to merge.

**Pull request merged**